### DISCUSSION TITLE
Clarifications on Users PII Data

### DISCUSSION THREAD

**Discussion by Shakthieshwari** [Sunbird-Obsrv/Community#12] [Query]
Hello Team,

In the Portal -> Under Course Progress Exhaust Page, We can get User Exhaust which contains User's PII Data.

Similar to this, Under the Program Dashboard now we should get User Exhaust which contains PII Data for all the projects/observations resource

Jira Ticket Link :- https://project-sunbird.atlassian.net/browse/SB-30235

@sowmya-dixit Can you please help us out, on how to get this data ?

Cc- @aks30, @kiranharidas187, @vijiurs

Thanks

**Comment by Shakthieshwari** [comment]
Team, Any update on this please ?

**Reply by rhwarrier** [reply]
@Shakthieshwari  
The user PII is made available to the Course mentor after obtaining explicit consent from the user regarding the sharing of their data. Is a similar approach being followed for Programs? Data cannot be shared without consent. 

Please discuss the approach with DC / @anandp504 and get to an agreement before you start to build.
@sowmya-dixit please provide inputs as necessary.

**Reply by Shakthieshwari** [reply]
yes @rhwarrier . We are explicitly getting consent from the users regarding the sharing of their data.

Sure will discuss with DC and then get the approval.

Thanks

**Reply by rhwarrier** [reply]
@Shakthieshwari  I can see that the ticket mentions consent as well - assuming you work with SB Ed for the front end changes for capturing user consent for Programs. Please review the approach with DC and close before you take up dev work. Thanks .

**Reply by Shakthieshwari** [reply]
@rhwarrier @vrayulu @santhosh V Please let me know the good time to connect for DC Approval on this story?

And also we wanted to know how we will get email id and mobile no for a user as it is encrypted in the user API Response, If we get know the approach followed for User Exhaust in the Course Progress, may be we can implement the same here as well in program dashboard.

Thanks 

Cc- @aks30, @kiranharidas187, @vijiurs

**Reply by pramodkvarma** [reply]
I looked at JIRA story and it doesn't explain the actual need for program manager to see everyone's phone and email. @Shakthieshwari can you please help me understand? What does she/he do seeing a list of people's email/phone? I am curious.

**Reply by Shakthieshwari** [reply]
@Prateek-slokam @aishwaryashikshalokam Can you please answer @pramodkvarma Query?

**Reply by aishwaryashikshalokam** [reply]
Dear Pramod 

You are right! The program manager does not need an Email Id or Phone No. as per our use case. We used an existing template of course PII data and hence the confusion. 
We wanted to understand if we just provide Names in the reports, will it still require the user's consent? 

The following fields are actually needed:  Program Name, Program Id, Resource Name, Resource Id, UUID, User Name, State, District, Org Name, Block Name, Cluster Name, UserType, User Subtype, School name, School Id

We just noticed that current administrator guidelines say that sharing of Names of registered users does not require consent.
Use Case:
- state officials ( Program managers) want to recognize and appreciate users who have consumed resources in the desired manner. Eg: Completing observations during a given duration, taking up and completing improvement projects, etc.

**Reply by pramodkvarma** [reply]
Thanks @aishwaryashikshalokam for clarifying. Good to know the conclusion and I am in agreement. Showing name, in general, should be fine unless the name is so unique the real person can be identified just by name. 

But again I urge you to ask what's the purpose of having name of user in that report? What does the manager who is looking at report do seeing the names? It's important that showing a field, especially user's personal data, should have a clear purpose. What happens if names are not there, what capability will be lost?

**Reply by aishwaryashikshalokam** [reply]
Dear @mmanimegalai

As per Pramod's comment Name along with the mentioned fields (  Program Name, Program Id, Resource Name, Resource Id, UUID, User Name, State, District, Org Name, Block Name, Cluster Name, UserType, User Subtype, School name, School Id) will not be considered as a PII.

In our call, I understood that Name in combination with the above-mentioned fields will be. 
Can you please clarify so that we can proceed further?

cc: @prateek-slokam, @kameshbhr, @varsha-sl

**Reply by kameshbhr** [reply]
@aishwaryashikshalokam Like Pramod mentioned:
1. Name in general is not considered as a PII
2. You should be clear on why the name is required in the report

Based on your previous comment, if the Name is required so that the Program managers can recognize and reward the users, it means they will be able to identify the users with the provided information in the report. So it become a PII.

From our previous experience with Courses, here is what I suggest:
1. Have one report with UUID, State, District etc. (without Name). This can be used for deriving insights about the program execution
2. Have another report which has UUID, Name and any other user information as required. This can be used for identifying the users to provide rewards and recognition.
For populating the information in the second report, take an explicit user consent. Report should contain the information of only the users who have provided consent.

**Reply by aishwaryashikshalokam** [reply]
Thank you @kameshbhr , this helps!
